controllers/applicationsnapshotenvironmentbinding_controller.go

```diff
@@ -51,12 +52,21 @@ import (
 // SnapshotEnvironmentBindingReconciler reconciles a SnapshotEnvironmentBinding object
 type SnapshotEnvironmentBindingReconciler struct {
 	client.Client
-	Scheme       *runtime.Scheme
-	Log          logr.Logger
-	AppFS        afero.Afero
-	Generator    gitopsgen.Generator
-	GitHubClient *gh.Client
-	GitToken     string
+	Scheme             *runtime.Scheme
+	Log                logr.Logger
+	AppFS              afero.Afero
+	Generator          gitopsgen.Generator
+	GitHubClient       *gh.Client
+	GitToken           string
+	GitOpsJobNamespace string
+
+	// DoLocalGitOpsGen determines whether or not to only spawn off gitops generation jobs, or to run them locally inside HAS. Defaults to false
+	DoLocalGitOpsGen bool
+
+	// AllowLocalGitopsGen allows for certain resources to generate gitops resources locally, *if* an annotation is present on the resource. Defaults to false
+	AllowLocalGitopsGen bool
+
+	GitOpsJobClientSet *kubernetes.Clientset
 }
 
 //+kubebuilder:rbac:groups=appstudio.redhat.com,resources=snapshotenvironmentbindings,verbs=get;list;watch;create;update;patch;delete
```

Comment:
do we need to mention the job resource and its verbs here

Reply:
Not strictly necessary since the component controller specifies them, but I'll add them here to just to be consistent.

Reply:
Fixed